### PR TITLE
Split area

### PR DESCRIPTION
**Pull request by JasonMarechal25**

### CODE REVIEW

File: src/libs/antares/CMakeLists.txt [hunk 1]
```diff
@@ -533,6 +509,7 @@ add_library(libantares-core
 	${SRC}
 	${SRC_LOGS}
 	${SRC_MEMORY}
+	${SRC_OBJECT}
```
Comment (flomnes) [inline src/libs/antares/CMakeLists.txt]: Please remove this

File: src/ui/simulator/toolbox/input/renewable-cluster.h [hunk 2]
```diff
@@ -42,6 +42,7 @@
 #include "../components/htmllistbox/item/renewable-cluster-item.h"
 #include "../components/htmllistbox/datasource/renewable-cluster-order.h"
 #include "../components/htmllistbox/item/renewable-cluster-item.h"
+#include "area.h"
```
Comment (flomnes) [inline src/ui/simulator/toolbox/input/renewable-cluster.h:45]: Unnecessary ?

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/ui/simulator/toolbox/input/renewable-cluster.h (hunk 2 above)
Comment (JasonMarechal25) [reply src/ui/simulator/toolbox/input/renewable-cluster.h:45]: It's area.h of toolbox, not of study

File: src/libs/antares/study/area/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+source_group("study\\area" FILES ${SRC_AREAS})
+
+add_library(area
+        ${SRC_AREAS})
+
+target_include_directories(area
+        PRIVATE
+        ${CMAKE_SOURCE_DIR}/ext/yuni/src
+        ${CMAKE_SOURCE_DIR}/solver/writer
+        ${CMAKE_SOURCE_DIR}/libs
```
Comment (sylvlecl) [inline src/libs/antares/study/area/CMakeLists.txt:25]: These are cyclic dependencies:
the new area lib depends on includes from downstream libs (for example study.h) which themselves depend on the area lib.

We must avoid this, it defeats the purpose of having smaller libraries with clear dependencies.
(in particular, changing study.h will still trigger the rebuild of the area library).

So, we should either:
- not separate this lib
- separate it properly, with removal of reverse dependencies

We must take care of libraries depending **only** on headers from upstream libraries.
We need to find what's the best way to ensure this with cmake. In the best case, headers should come simply through the definition of linked libraries. cmake 3.23 introduced the feature of "files set" which allows to define properly public headers, it seems it was not so clear before that version. To be investigated !

**Review by sylvlecl — CHANGES_REQUESTED** [review]
See comment: I don't think we should merge this in the current state, this introduces cyclic dependencies between libraries.
We need to be consistent between linked libraries and included headers. We need to identify best way to achieve this with cmake.

**Comment by JasonMarechal25** [comment]
Closing to prevent conflict with existing PRs and redo following new guidelines

**Pull request closed without merging**